src/Asset/CacheEntrypointLookup.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/*
+ * This file is part of the Symfony WebpackEncoreBundle package.
+ * (c) Fabien Potencier <fabien@symfony.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\WebpackEncoreBundle\Asset;
+
+class CacheEntrypointLookup extends EntrypointLookup
+{
+    private $entrypoints;
+
+    public function __construct(array $entrypoints)
+    {
+        $this->entrypoints = $entrypoints;
+    }
+
+    protected function getEntriesData(): array
+    {
+        return $this->entrypoints;
+    }
```

Comment:
Hmm. Unfortunately, making this protected isn't the best way to make this extension point - though it was the easiest! ;) It forces us to make the method protected... and it also forces us to remove the `@final` on the parent class (which we would like to not need to do).

I see why you did it - there is actually a decent amount of logic in `EntrypointLookup` other than just `json_decode()` on the `manifest.json` file. 

I have a different idea. Sorry for the back-and-forth - this isn't an area that I usually touch in the Symfony core :). Here's my idea:

1) Let's *always* cache. Basically, we will move the caching logic into EntrypointLookup and not make it optional
2) But, we will not use, for example, the `cache.system` cache or something like that. Instead, there is a system in the `Config` component of Symfony for this type of caching - an object called the `ConfigCacheFactory`.

Basically, we use the `ConfigCacheFactory` to cache the results of parsing the `entrypoints.json` file. But, built into that system is a mechanism that will automatically recache it if the file modified time of `entrypoints.json` is updated (in dev only, of course). It means that, in dev, the cache is loaded the first time you access the page, then reloaded whenever the file changes. In `prod`, we would create a new "cache warmer" so that it's loaded during `cache:warmup`.

A good example of this system is the translator: 

* https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/Resources/config/translation.xml#L19-L21
* https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Translation/Translator.php#L323-L327

Can you try this out? I think it's ultimately the best solution.

Thanks!